svm/src/account_loader.rs

```diff
@@ -167,35 +178,85 @@ pub(crate) fn load_accounts<CB: TransactionProcessingCallback>(
 ) -> Vec<TransactionLoadResult> {
     txs.iter()
         .zip(validation_results)
-        .map(|etx| match etx {
-            (tx, Ok(tx_details)) => {
-                // load transactions
-                load_transaction_accounts(
-                    callbacks,
-                    tx,
-                    tx_details,
-                    error_metrics,
-                    account_overrides,
-                    feature_set,
-                    rent_collector,
-                    loaded_programs,
-                )
-            }
-            (_, Err(e)) => Err(e),
+        .map(|(transaction, validation_result)| {
+            load_transaction(
+                callbacks,
+                transaction,
+                validation_result,
+                error_metrics,
+                account_overrides,
+                feature_set,
+                rent_collector,
+                loaded_programs,
+            )
         })
         .collect()
 }
 
+fn load_transaction<CB: TransactionProcessingCallback>(
+    callbacks: &CB,
+    message: &impl SVMMessage,
+    validation_result: TransactionValidationResult,
+    error_metrics: &mut TransactionErrorMetrics,
+    account_overrides: Option<&AccountOverrides>,
+    feature_set: &FeatureSet,
+    rent_collector: &RentCollector,
+    loaded_programs: &ProgramCacheForTxBatch,
+) -> TransactionLoadResult {
+    match validation_result {
+        Err(e) => TransactionLoadResult::NotLoaded(e),
+        Ok(tx_details) => {
+            let load_result = load_transaction_accounts(
+                callbacks,
+                message,
+                &tx_details,
+                error_metrics,
+                account_overrides,
+                feature_set,
+                rent_collector,
+                loaded_programs,
+            );
+
+            match load_result {
+                Ok(loaded_tx_accounts) => TransactionLoadResult::Loaded(LoadedTransaction {
+                    accounts: loaded_tx_accounts.accounts,
+                    program_indices: loaded_tx_accounts.program_indices,
+                    fee_details: tx_details.fee_details,
+                    rent: loaded_tx_accounts.rent,
+                    rent_debits: loaded_tx_accounts.rent_debits,
+                    rollback_accounts: tx_details.rollback_accounts,
+                    compute_budget_limits: tx_details.compute_budget_limits,
+                    loaded_accounts_data_size: loaded_tx_accounts.loaded_accounts_data_size,
+                }),
+                Err(err) => TransactionLoadResult::FeesOnly(FeesOnlyTransaction {
```

Comment:
This is where we convert load failures into fees-only transactions

Reply:
txs that cannot pay fees would still have an error here. Do we not need to filter out the fee-payer errors here to make sure we can still take fees?

Reply:
Fee payer validation is done before the load transaction accounts phase of the tx pipeline. Note how above we turn validation errors into the NotLoaded variant.

Reply:
Ah yeah I see it now